### PR TITLE
v0.7

### PR DESCRIPTION
**Pull request by HarryET**
# Description

This release will bump the versions of dependencies as well as bring the codebase into a modern rust version. Merging this PR is reliant on https://github.com/WalletConnect/hyper-alpn/pull/9 getting merged and released first.

## How Has This Been Tested?

Using `cargo test`

## Due Diligence

* [x] Breaking change
* [x] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,24 +13,25 @@ keywords    = ["apns", "apple", "push", "async", "http2"]
...
-hyper-alpn = "0.3"
+tracing = { version = "0.1.36", optional = true }
 
 [dev-dependencies]
 argparse = "0.2"
-pretty_env_logger = "0.4"
-indoc = "1"
+tracing-subscriber = "0.3.15"
 tokio = { version = "1", features = ["rt-multi-thread", "macros"] }
+hyper = { version = "0.14", features = ["client", "http2", "tcp"] }
```
Comment (arein) [inline Cargo.toml:37]: why is this only a dev dependency now?

**Review by arein — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (HarryET) [reply Cargo.toml:37]: it's both but we only need the `tcp` feature in dev for the examples

Comment (pimeys) [reply Cargo.toml:37]: You might want to use the new resolver. The old one will use the largest set of features for both builds (a cargo bug).

Comment (HarryET) [reply Cargo.toml:37]: We have moved to `edition = 2021` which implies `resolver = 2` according to docs

**Comment by HarryET** [comment]
This PR is waiting for @jrconlin's PR from the Mozilla Service's repo before we cary on so that I'm not writing the same fixes they are.

**Comment by jrconlin** [comment]
you could probably land this one and I can merge the changes in `localized.rs` and the others. This looks like it's mostly updating code. 

(Sorry, been busy with other projects. Will try to get the PR done this week.)

**Comment by HarryET** [comment]
> you could probably land this one and I can merge the changes in `localized.rs` and the others. This looks like it's mostly updating code.
> 
> (Sorry, been busy with other projects. Will try to get the PR done this week.)

👍🏼  Also looking at #62 to try get this into v0.7, will double check you changes and if non breaking I'll merge this and just release a v0.7.1 afterwards with you changes.

File: Cargo.toml [hunk 2]
```diff
@@ -13,24 +13,29 @@ keywords    = ["apns", "apple", "push", "async", "http2"]
...
-futures = "0.3"
+openssl = { version = "0.10", optional = true }
+hyper = { version = "0.14", default-features = false, features = ["client", "http2"] }
+hyper-alpn = "0.4"
 http = "0.2"
 base64 = "0.13"
-log = "0.4"
-hyper = { version = "0.14", features = ["client", "http2", "tcp"] }
-hyper-alpn = "0.3"
+tracing = { version = "0.1.36", optional = true }
```
Comment (heilhead) [inline Cargo.toml:33]: Can we avoid the patch version here and in other dependencies? e.g. `0.1.36` -> `0.1`. To give more flexibility for the consumer to resolve compatible versions without duplicates.

**Review by xav — APPROVED** [review]
Since this is OSS, can we add a `.editorconfig` while we're at it?

**Comment by HarryET** [comment]
> Since this is OSS, can we add a `.editorconfig` while we're at it?

I'll make an issue and add this later

**Review by heilhead — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (HarryET) [reply Cargo.toml:33]: 👍🏼 will change before merge

**Pull request merged**